effectsmanager.h

```diff
@@ -56,30 +59,31 @@ class EffectsManager : public IEffectsManager
         return _effects.size();
     }
 
-    vector<reference_wrapper<ILEDEffect>> Effects() const override
+    vector<shared_ptr<ILEDEffect>> Effects() const override
     {
-        vector<reference_wrapper<ILEDEffect>> effects;
-        effects.reserve(_effects.size());
-        for (auto &effect : _effects)
-            effects.push_back(*effect);
-        return effects;
+        lock_guard lock(_effectsMutex);
+        return _effects;
     }
 
     // Add an effect to the manager
-    void AddEffect(unique_ptr<ILEDEffect> effect) override
+    void AddEffect(shared_ptr<ILEDEffect> effect) override
     {
+        lock_guard lock(_effectsMutex);
+
         if (!effect)
             throw invalid_argument("Cannot add a null effect.");
-        _effects.push_back(std::move(effect));
+        _effects.push_back(effect);
```

Comment:
I might need schooling on whether it's useful to std::move a shared pointer or to let the operator just do the refcount, etc...

Reply:
You can std::move a shared pointer just like anything else. You just have to be sure the instance you're moving from will no longer be used... like with anything else.

From an actual benefit perspective it's a different matter. With a shared pointer being a pointer and a ref counter (yeah, I know I'm oversimplifying) the question is how much you really gain with it.